packngo.go

```diff
@@ -230,6 +231,45 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 	return &response, err
 }
 
+// dumpDeprecation logs headers defined by
+// https://tools.ietf.org/html/rfc8594
+func dumpDeprecation(resp *http.Response) {
+	deprecation := resp.Header.Get("Deprecation")
+
+	links := resp.Header.Values("Link")
+	link := map[string]string{}
+
+	// TODO multiple relevant links would be reduced to one
+	for i := range links {
+		if strings.Contains(links[i], "rel=\"sunset\"") {
+			link["sunset"] = strings.Split(links[i], ";")[0]
+		}
+		if strings.Contains(links[i], "rel=\"deprecation\"") {
+			link["deprecation"] = strings.Split(links[i], ";")[0]
+		}
+	}
+	if len(link) == 0 {
+		link["sunset"] = strings.Join(links, ", ")
+		link["deprecation"] = link["sunset"]
+	}
+	uri := ""
+	if resp.Request != nil {
+		uri = resp.Request.Method + " " + resp.Request.RequestURI
+	}
+
+	if deprecation != "" {
+		link := link["deprecation"]
+
+		log.Printf("WARNING: %q is deprecated since %s %s", uri, deprecation, link)
```

Comment:
I'm not sure about the scope of API deprecations. Could `POST /projects/{id}/devices {"plan": "t1.small"}` report a plan deprecation? An image deprecation? In those cases `"POST /projects/{id}/devices" is deprecated` would be the wrong message to send.

I originally worded this as `"POST ..." reported a deprecation` but I struggled with how to generically phrase this and include the date.

Perhaps: `"POST ..." included a deprecation, effective {deprecation header contents} <link url>`

Reply:
I think `POST to ... reported a {deprecation,sunset} ... ` is the best wording.

I wasn't sure if printing deprecations straight up to stdout/stderr is a good idea, but thinking it through, I didn't come up with a better way to show it to a user. I was thinking to involve an envvar (allowing or suppressing the deprecation messages) or conditioning it with PACKNGO_DEBUG (but that will hide amongst a lot of other debugging output).

Reply:
Perhaps we need a Client method or property that can squelch these messages `.LogDeprecations = false` / `.SetDeprecationLogging(false)`
I noticed that this is the only place outside of tests that will emit a log message, so I have the same reluctance. Perhaps:

```
PACKNGO_LOG=/dev/null # default /dev/stderr, can be redirected elsewhere
```

We could also offer a client accessor to fetch deprecation messages, either from the last request or all messages in the client lifetime.

Reply:
I think plain printing is OK by now. We can fix it later if we notice that it's not clear enough, or if it's disturbing the users.